lib/reek/code_comment.rb

```diff
@@ -94,16 +95,20 @@ def initialize(detector_name:, original_comment:, line:, source:, options: {})
         @line             = line
         @source           = source
         @options          = options
+        @detector_class   = nil # We only know this one after our first initial checks
+        @parsed_options   = nil # We only know this one after our first initial checks
```

Comment:
Fully agree! I had the exact same thoughts. My initial idea was to create separate classes for each validator, but since the validations are dependent on each other this lead to a huge amount of boilerplate code where we would do nothing else but pass parameters around.

I suggest to keep it like that because there is also one last validation missing: the ones for the datatypes. When we have that in place we're fully done validating source code comments (yay!) and then we have the test coverage in place to refactor.